sentry-ruby/lib/sentry/transport/http_envelope_transport.rb

```diff
@@ -0,0 +1,24 @@
+require 'faraday'
+
+module Sentry
+  class HTTPEnvelopeTransport < HTTPTransport
+    attr_accessor :conn, :adapter
+
+    def initialize(*args)
+      super
+      @endpoint = @dsn.envelope_endpoint
+    end
+
+    def encode(event_hash)
+      event_id = event_hash[:event_id] || event_hash['event_id']
+
+      envelope = <<~ENVELOPE
+        {"event_id":"#{event_id}","dsn":"#{configuration.dsn.to_s}","sdk":#{Sentry.sdk_meta.to_json},"sent_at":"#{DateTime.now.rfc3339}"}
+        {"type":"event","content_type":"application/json"}
```

Comment:
Here we need to check what type of event comes in and set `transaction` for a transaction once we add tracing support.

Reply:
sure I'm aware of that 👍